fusillade/api/__init__.py

```diff
@@ -0,0 +1,202 @@
+import collections
+import functools
+import logging
+import os
+import re
+
+import chalice
+import requests
+
+from connexion import FlaskApp
+from connexion.decorators.validation import ParameterValidator, RequestBodyValidator
+from connexion.resolver import RestyResolver
+
+from fusillade import FusilladeBindingException, Config
+
+
+class FusilladeApp(FlaskApp):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+class FusilladeParameterValidator(ParameterValidator):
+    """
+    The ParameterValidator provided by Connexion immediately returns a value if the validation fails.  Therefore, our
+    code is never invoked, and the common_error_handler in the connexion.App object is never called.  This means error
+    messsages are not returned using our standard error formats.
+
+    The solution is to trap the validation results, and if it fails, exit the validation flow.  We catch the exception
+    at the top level where the various validators are called, and return a value according to our specs.
+    """
+
+    @staticmethod
+    def validate_parameter(*args, **kwargs):
+        result = ParameterValidator.validate_parameter(*args, **kwargs)
+        if result is not None:
+            raise FusilladeBindingException(result)
+        return result
+
+    def __call__(self, function):
+        origwrapper = super().__call__(function)
+
+        @functools.wraps(origwrapper)
+        def wrapper(request):
+            try:
+                return origwrapper(request)
+            except FusilladeBindingException as ex:
+                return ex.to_problem()
+
+        return wrapper
+
+
+class FusilladeRequestBodyValidator(RequestBodyValidator):
+    """
+    The RequestBodyValidator provided by Connexion immediately returns a value if the validation fails.  Therefore, our
+    code is never invoked, and the common_error_handler in the connexion.App object is never called.  This means error
+    messsages are not returned using our standard error formats.
+
+    The solution is to trap the validation results, and if it fails, exit the validation flow.  We catch the exception
+    at the top level where the various validators are called, and return a value according to our specs.
+    """
+
+    def validate_schema(self, *args, **kwargs):
+        result = super().validate_schema(*args, **kwargs)
+        if result is not None:
+            raise FusilladeBindingException(result.body['detail'])
+        return result
+
+    def __call__(self, function):
+        origwrapper = super().__call__(function)
+
+        @functools.wraps(origwrapper)
+        def wrapper(request):
+            try:
+                return origwrapper(request)
+            except FusilladeBindingException as ex:
+                return ex.to_problem()
+
+        return wrapper
+
+
+class ChaliceWithConnexion(chalice.Chalice):
```

Comment:
This code follows the same pattern as https://github.com/HumanCellAtlas/query-service/blob/master/dcpquery/api/__init__.py